src/electron/platform/android/unified-result-builder.ts

```diff
@@ -34,14 +34,14 @@ export const createBuilder = (
     const payload: UnifiedScanCompletedPayload = {
         scanResult: getUnifiedResults(scanResults, ruleInformationProvider, uuidGenerator),
         rules: getUnifiedRules(scanResults, ruleInformationProvider, uuidGenerator),
-        platformInfo: getPlatformData(scanResults),
+        platformInfo: getPlatformData(scanResults) ?? undefined,
```

Comment:
What's the reason we don't want to possibly return undefined from getPlatformData?

Reply:
In general, my preference is to use `null` over `undefined` to represent "we explicitly decided to return a non-value here", which is what `getPlatformData`'s `null` case represents. I decided to keep its current implementation (which already returned `null` for this case) based on a combination of that preference and minimizing the change size.

The obvious followup question is "if `null` makes more sense here why not update `UnifiedScanCompletedPayload` to use that instead?" I debated how far to go here with myself for quite some time. Ultimately, there's not a lot of practical difference for this case (or most of our cases) in using `null` vs `undefined` (we typically check for it using `!= null`, which treats the cases identically), and all of the `unified-data-interface` types use `property?: Type` instead of `property: Type | null` throughout, so the options would have been either "do it for just the `platformInfo` property, but then it's inconsistent", "do it for all the properties, but then it's a huge PR for negligible value", or "just do the conversion here". I opted for the latter as the best of bad options.